src/test/java/http_server/ServerTest.java

```diff
@@ -0,0 +1,84 @@
+package http_server;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.*;
+
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.Assert.assertEquals;
+
+public class ServerTest {
+
+    private Server server;
+    private ServerSocketSpy.ClientSocketSpy clientSocketSpy;
+    private ByteArrayOutputStream outputStream;
+    private ServerSocketSpy serverSocketSpy;
+
+    @Before
+    public void setUp() {
+        server = new Server("localhost", 5000);
+        outputStream = new ByteArrayOutputStream();
+        ByteArrayInputStream inputStream = new ByteArrayInputStream("".getBytes());
+        clientSocketSpy = new ServerSocketSpy.ClientSocketSpy(inputStream, outputStream);
+        serverSocketSpy = new ServerSocketSpy(clientSocketSpy);
+    }
+
+    @Test
+    public void testsAConnectionCanBeMade() {
+        server.run(serverSocketSpy);
+
+        assertTrue(serverSocketSpy.acceptWasCalled);
+    }
+
+    @Test
+    public void hasAHost() {
+        Server server = new Server("localhost", 5000);
```

Comment:
Why are you doing this again?  Should you not be able to use the server initialized in setUp?